data/txHandler.go

```diff
@@ -224,6 +242,18 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 	dec := protocol.NewMsgpDecoderBytes(rawmsg.Data)
 	ntx := 0
 	unverifiedTxGroup := make([]transactions.SignedTxn, 1)
+
+	// consume a capacity unit
+	capguard, err := handler.erl.ConsumeCapacity(rawmsg.Sender.(util.ErlClient))
+	if err != nil {
+		logging.Base().Warnf("Could not consume capacity for client %v: %v", rawmsg.Sender, err)
+		return network.OutgoingMessage{Action: network.Ignore}
+	}
+	// if the backlog Queue has 50% of its buffer back, turn congestion control off
+	if float64(cap(handler.backlogQueue))*0.5 > float64(len(handler.backlogQueue)) {
```

Comment:
The ERL has no concept of the underlying resource it is protecting, so it would need to be newly informed of `backlogQueue`. Since the handler owns the backlog queue, it makes sense that the handler would also own the management (congestion on/off) on that queue.

What data would you suggest passing to the ERL so it can do this work itself? The whole backlog queue? Just the ratio?

Reply:
You know the size of `backlogQueue` via the `len(sharedCapacity)`.

```go
// this can be done inside the erl..
if float64(cap(erl.sharedCapacity))*0.5 > float64(erl(handler.sharedCapacity)) {
```

Reply:
Ah yea! I had hadn't considered them as explicitly the same value before, but they certainly are (or should be).

Reply:
(offline discussion: we'll leave this exterior toggle control in place because it is not clear that all consumers would want identical toggle control, and this leaves things more individually unit-testable.)

Reply:
The backlog queue size is increased with this change, but not on this line.

Before it was "approximately one block of transactions"

With this PR it is (approximately two blocks of transactions) + (20 capacity per peer * max inbound connections)